File: X/python/XRPL_UKL1/soal7_1.py
#!/bin/python3
"""
Soal        : 7
Modelsoal   : 2
"""

def biaya(kategorii, jalurr, bulann): #No switch case ?
    if(jalurr == "SBMPTN"):
        if(kategorii == "A"):
            dsp = 5000000
            spp = 500000
        elif (kategorii=="B"):
            dsp = 15000000
            spp = 1000000
        elif(kategorii == "C"):
            dsp = 30000000
            spp = 2000000

    elif(jalurr=="SNMPTN"):
        if(kategorii=="A"):
            dsp = 7000000
            spp = 500000
        elif(kategorii=="B"):
            dsp = 17000000
            spp = 1000000
        elif(kategorii=="C"):
            dsp = 35000000
            spp = 2000000

    elif(jalurr=="Mandiri"):
        if(kategorii=="A"):
            dsp = 10000000
            spp = 1000000
        elif(kategorii=="B"):
            dsp = 25000000
            spp = 2000000
        elif(kategorii=="C"):
            dsp = 50000000
            spp = 3000000

    return dsp+(spp*bulann)

nama=["Mira","Nina","Oemar","Pena"]
jalur=["SBMPTN","SNMPTN","Mandiri","SBMPTN"]
alamat=["Sawojajar","Kedung Kandang","Ijen","Dinoyo"]

id=int(input("masukkan id mahasiswa       : "))-1
#id=2-1
if(id<0 | id>len(nama)):
    print("Invalid input\n");exit(0)

ortu=int(input("masukkan pendapat ortu      : "))
bulan=int(input("masukkan jumlah bulan spp   : "))
#ortu=20000
#bulan=2

if (ortu >10000000):
    kategori = "C"
elif (ortu >=2000000):
    kategori = "B"
elif (ortu <=2000000):
    kategori = "A"

print(
f"""
=================================
Id mahasiswa                    : {id+1}
Nama mahasiswa                  : {nama[id]}

Jalur masuk                     : {jalur[id]}
Kategori pemdapatan             : {kategori}
Jumlah biaya                    : {biaya(kategori,jalur[id],bulan)}
Alamat                          : {alamat[id]}
=================================
""")

File: X/javascript/XRPL_UKL1/soal7_1.js
/*
Soal        : 7
Modelsoal   : 2
*/
const readline = require("readline");

const input = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

opsi_biaya = {
    "SBMPTN": {
        "A": { dsp: 5_000_000, spp: 500_000 },
        "B": { dsp: 15_000_000, spp: 1_000_000 },
        "C": { dsp: 3_000_0000, spp: 2_000_000 },
    },
    "SNMPTN": {
        "A": { dsp: 7_000_000, spp: 5_00000 },
        "B": { dsp: 17_000_000, spp: 1_000_000 },
        "C": { dsp: 35_000_000, spp: 2_000_000 },
    },
    "Mandiri": {
        "A": { dsp: 1_000_0000, spp: 1_000_000 },
        "B": { dsp: 25_000_000, spp: 2_000_000 },
        "C": { dsp: 5_000_0000, spp: 3_000_000 },
    },
};

function biaya(kategori, jalur, bulan) {
    total = 0;
    const options = opsi_biaya[jalur]?.[kategori];
    if (options) {
        var { dsp, spp } = options;
        total = dsp + spp * bulan;
    }
    return total;
}

nama = ["Mira", "Nina", "Oemar", "Pena"],
    jalur = ["SBMPTN", "SNMPTN", "Mandiri", "SBMPTN"],
    alamat = ["Sawojajar", "Kedung Kandang", "Ijen", "Dinoyo"],

    input.question("masukkan id mahasiswa       : ", function (id) {
        id = parseInt(id) - 1;
        if (id < 0 || id > nama.length) {
            console.log("Invalid input"); process.exit(0);
        }

        input.question("masukkan pendapat ortu      : ", function (ortu) {
            input.question("masukkan jumlah bulan spp   : ", function (bulan) {
                ortu = parseInt(ortu);
                bulan = parseInt(bulan);

                if (ortu > 1_000_0000) {
                    kategori = "C";
                }
                else if (ortu >= 2_000_000) {
                    kategori = "B";
                }
                else if (ortu <= 2_000_000) {
                    kategori = "A";
                }
                input.close();

                console.log(`
=================================
Id mahasiswa                    : ${id + 1}
Nama mahasiswa                  : ${nama[id]}

Jalur masuk                     : ${jalur[id]}
Kategori pemdapatan             : ${kategori}
Jumlah biaya                    : ${biaya(kategori, jalur[id], bulan)}
Alamat                          : ${alamat[id]}
=================================
`);
            });
        });
    });
